magento/magento2
# Add new GraphQL query for single product retrieval

### Description (*)
This PR adds a new GraphQL endpoint for the products fetching by IDs.
The new GraphQL single product retrieval endpoint was requested at https://github.com/magento/catalog-storefront/issues/12 issue.
After the conversations in this PR, there was a request to support the fetching of several products by IDs. [Comment Link](https://github.com/magento/magento2/pull/28316#discussion_r438195391)

### Fixed Issues (if relevant)

1. Fixes magento/catalog-storefront#12: New GraphQL endpoint for single product retrieval

### Manual testing scenarios (*)

1. Create a product
2. Use a similar query to retrieve the products data by IDs
```
{
  productsByID(ids: [1, 2]) {
    id
    name
    sku
    categories {
      id
      name
    }
    media_gallery {
      url
    }
  }
}
```

The result should be like on the following screenshot:
<img width="1381" alt="Screenshot 2020-06-24 at 09 30 37" src="https://user-images.githubusercontent.com/31502344/85508789-65f30480-b5fd-11ea-848f-b0e258946a69.png">


### Contribution checklist (*)
